asmcomp/arm64/emit.mlp

```diff
@@ -327,7 +327,9 @@ let float_literal env fl =
 let emit_literals env =
   if env.float_literals <> [] then begin
     if macosx then
-      `	.section	__TEXT,__literal8,8byte_literals\n`;
+      `	.section	__TEXT,__literal8,8byte_literals\n`
```

Comment:
I *think* the analog of `.rodata` in OS X is `__DATA,__const`.

Reply:
Pinging @tmcgilchrist, can you try this on your machine eventually?

Reply:
What OCaml programs are you using to test this against? It'd be useful to verify the assembly / binary created to validate how segments/sections get put together.

The current `__TEXT,__literal8` looks correct and I would leave it as is. `__DATA,__const` seems to be more for pointers to constant strings or pointers to constants that need fixups during load.

| Segment and section name | Contents | 
|--|--|
| __TEXT,__literal8 | 8-byte literal values. The compiler places double-precision floating point constants in this section. The static linker coalesces these values, removing duplicates, when building the final product. With some architectures, it’s more efficient for the compiler to use immediate load instructions rather than adding to this section. |
| __DATA,__data| Initialized mutable variables, such as writable C strings and data arrays. |
|__DATA,__const | Initialized relocatable constant variables.|

taken from *OS X ABI Mach-O File Format Reference* https://github.com/aidansteele/osx-abi-macho-file-format-reference

> In order to improve performance and security the static linker (ld) now moves globals that are marked as constant into a new segment: __DATA_CONST. These globals may consist of compiler generated pointers that the dynamic linker (dyld) needs to fix up during load, but are otherwise constant such as vtables and explicitly declared constant pointers. Once dyld has finished loading the image it makes __DATA_CONST readonly.

```c
static int value1 = 0;                  // Stored in __DATA
static int value2 = 0;                  // Stored in __DATA


const int * const valuePtr = &value1;   // Stored in __DATA_CONST


// ERROR: Attempting to store a value to a constant pointer
(int *)valuePtr = &value2;
```
Referencing https://developer.apple.com/documentation/xcode-release-notes/xcode-11-release-notes?language=objc

Reply:
A good test for this change of section is `testsuite/tests/lib-float/test.ml` which has many floating-point constant literals.

But reading the documentation you've quoted (thanks!), I agree no macOS change is necessary, the existing logic is the most appropriate.

Reply:
For interest sake, changing to use `__DATA,__const` doesn't break the test suite and yields a small change in the `__TEXT` `__DATA` sections on MacOS for `testsuite/tests/lib-float/test.ml` :
```shell
 $ size test-data-const.exe test.exe 
__TEXT	__DATA	__OBJC	others	dec	hex
573440	262144	0	4295327744	4296163328	100124000	test-data-const.exe
557056	245760	0	4295327744	4296130560	10011c000	test.exe
```
With load commands in the final binary being essentially the same (minus the size / addr values). See `otool -lv` for displaying the load commands.

Reply:
But since the linker is documented to merge identical constants when they are put in `__TEXT`, we should prefer that behaviour and not change anything on macOS.